pkg/controller/machine.go

```diff
@@ -793,6 +856,25 @@ func (c *controller) updateMachineStatus(
 	return clone, nil
 }
 
+// isMachineStatusEqual checks if the status of 2 machines is similar or not.
+func (c *controller) isMachineStatusEqual(s1, s2 v1alpha1.MachineStatus) bool {
+	tolerateTimeDiff := 30 * time.Minute
+	s1Copy, s2Copy := s1.DeepCopy(), s2.DeepCopy()
+	s1Copy.LastOperation.Description, s2Copy.LastOperation.Description = "", ""
+
+	if (s1Copy.LastOperation.LastUpdateTime.Time.After(time.Now().Add(tolerateTimeDiff))) || (s2Copy.LastOperation.LastUpdateTime.Time.After(time.Now().Add(tolerateTimeDiff))) {
+		return false
+	}
+	s1Copy.LastOperation.LastUpdateTime, s2Copy.LastOperation.LastUpdateTime = metav1.Time{}, metav1.Time{}
+
+	if (s1Copy.CurrentStatus.LastUpdateTime.Time.After(time.Now().Add(tolerateTimeDiff))) || (s2Copy.CurrentStatus.LastUpdateTime.Time.After(time.Now().Add(tolerateTimeDiff))) {
+		return false
+	}
+	s1Copy.CurrentStatus.LastUpdateTime, s2Copy.CurrentStatus.LastUpdateTime = metav1.Time{}, metav1.Time{}
```

Comment:
I don't understand this part. Why is this `if` block repeated twice? Also, is it possible to make the difference in timestamps using a single comparison instead of two?

Reply:
One if block is for `LastOperation`, and the other is for `CurrentStatus`.